### PR TITLE
Issue #37 fix fetch aws credentials on kubernetes

### LINKED ISSUE #37: Call to sts.getCredentials running from EKS does not pickup IRSA identity but EC2 identity
Our team has researched and we believe that the issue has something to do with the results shown below.

These two calls return different identities. When the `STS` object is instantiated without credentials, it picks up the IRSA identity. The problem occurs when passing credentials from `fetchAWSCredentials`, in that instance it picks up the ID of the worker node.

```js
const auth = await `fetchAwsCredentials()();
const sts = new AWS.STS({region: 'us-east-1', credentials: auth})
console.log(await sts.getCallerIdentity().promise())
```

```js
const sts = new AWS.STS({region: 'us-east-1'})
console.log(await sts.getCallerIdentity().promise())
```

Errors we're seeing in Splunk for our service look like this:

```json
{"error_id":"f586a0d5-0bad-4927-96bb-3763cd9be375","errors":[{"code":99216,"message":"The specified IAM principal is not valid."}]}
```

### PR DESCRIPTION
**Pull request by samarthmaniar**
Fixes #37 issue where sts gets incorrect identity when running in kubernetes cluster

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
+- 12
+- 14
 before_install:
 - sudo apt-get -qq update
 - sudo apt-get install -y oathtool
 jobs:
   include:
   - stage: publish code coverage
-    node_js: '8'
+    node_js: '12'
```
Comment (kyeotic) [inline .travis.yml:13]: we should run tests in the lowest version of node supported

Comment (samarthmaniar) [reply .travis.yml:13]: Sure. We will fix this.

Comment (samarthmaniar) [reply .travis.yml]: Fixed

**Comment by fieldju** [comment]
Why does this require a major version rev?
I am not seeing any code changes that are using features that are 12 specific, just that you are no longer building it on non-lts runtimes?

Do other npm libs such as react major version rev every time a non-lts runtimes rolls off?

**Comment by kyeotic** [comment]
@fieldju Its common for node runtime deprecation to be a major version bump, and this was done at my request.

**Pull request merged**